### PR TITLE
Test:  improve tests by checking error messages

### PR DESCRIPTION
**Pull request by dtivel**
Resolve https://github.com/dotnet/sign/issues/544.

This change simplifies tests a bit and checks error messages.

CC @erdembayar, @clairernovotny

### CODE REVIEW

**Comment by dtivel** [comment]
@erdembayar, can you take a look?

**Comment by erdembayar** [comment]
> @erdembayar, can you take a look?

Sure, I'll do today.

File: test/Sign.Cli.Test/Options/OptionTests.cs [hunk 1]
```diff
@@ -100,20 +111,29 @@ protected virtual void VerifyEqual(T? expectedValue, T? actualValue)
             Assert.Equal(expectedValue, actualValue);
         }
 
-        protected void VerifyHasError(string commandLine)
+        protected void VerifyHasErrors(string commandLine, params string[] expectedErrorMessages)
         {
             ParseResult result = Parse(commandLine);
+            HashSet<string> expectedMessages = new(expectedErrorMessages, StringComparer.Ordinal);
```
Comment (erdembayar) [inline test/Sign.Cli.Test/Options/OptionTests.cs:117]: Is there any case where ordering of error message matters? If yes, then we're losing ordering here.

File: test/Sign.Cli.Test/Options/MaxConcurrencyOptionTests.cs [hunk 2]
```diff
@@ -33,7 +35,9 @@ public void Option_WhenOptionIsMissing_HasDefaultValue()
         [InlineData(-1)]
         public void Option_WhenValueIsLessThanOne_HasError(int value)
```
Comment (erdembayar) [inline test/Sign.Cli.Test/Options/MaxConcurrencyOptionTests.cs:36]: Is there any limit for number of concurrent connections value? 
Anything more than 64 is not practical, right?

File: test/Sign.Cli.Test/Options/UriOptionTests.cs [hunk 3]
```diff
@@ -10,15 +10,29 @@ public abstract class UriOptionTests : OptionTests<Uri?>
...
         {
         }
 
         [Fact]
         public void Option_WhenValueFailsToParse_HasError()
         {
-            VerifyHasError("3");
+            const string value = "3";
+
+            if (Option.IsRequired)
```
Comment (erdembayar) [inline test/Sign.Cli.Test/Options/UriOptionTests.cs:23]: Just curious how this `IsRequired` option gets `true`, `false` values? Who is passing it?

**Review by erdembayar — APPROVED** [review]

File: test/Sign.Cli.Test/Options/OptionTests.cs (hunk 1 above)
Comment (dtivel) [reply test/Sign.Cli.Test/Options/OptionTests.cs:117]: Nope, ordering doesn't really matter.

File: test/Sign.Cli.Test/Options/UriOptionTests.cs (hunk 3 above)
Comment (dtivel) [reply test/Sign.Cli.Test/Options/UriOptionTests.cs:23]: `IsRequired` is set in the product and passed with the `Option` instance to the test.

https://github.com/dotnet/sign/blob/7b32935c42c720d96ceee799980a7f5ec2d2c296/src/Sign.Cli/CodeCommand.cs#L30-L32

File: test/Sign.Cli.Test/Options/MaxConcurrencyOptionTests.cs (hunk 2 above)
Comment (dtivel) [reply test/Sign.Cli.Test/Options/MaxConcurrencyOptionTests.cs:36]: Where did you get 64?  An upper bound doesn't appear to be officially documented, so I wouldn't want to hard-code a limit here.

Comment (erdembayar) [reply test/Sign.Cli.Test/Options/MaxConcurrencyOptionTests.cs:36]: I just throwed some random high numbers, maybe not important.

**Pull request merged**